Sources/FoundationEssentials/AttributedString/AttributedString+IndexValidity.swift

```diff
@@ -0,0 +1,93 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#if canImport(Synchronization)
+internal import Synchronization
+#endif
+
+extension AttributedString.Guts {
+    typealias Version = UInt
+    
+    #if canImport(Synchronization)
+    private static let _nextVersion = Atomic<Version>(0)
+    
+    static func createNewVersion() -> Version {
+        _nextVersion.wrappingAdd(1, ordering: .relaxed).oldValue
```

Comment:
Nit: I'd probably put the `#if canImport(Synchronization)` inside this function since that's implementation detail

Reply:
Sure I can do that if you prefer - I still would need a separate `canImport` check around the `private static let` variables since their types are different, but if you want I can split it into two `canImport` checks, one for around the variable and one within the body of the function